torch_xla/csrc/xla_graph_executor.cpp

```diff
@@ -856,6 +873,18 @@ void XLAGraphExecutor::ExtractIRAndPrepareXlaData_(
     torch::lazy::BackendDataPtr handle =
         WrapXlaData(xla::ComputationClient::Get()->CreateDataPlaceholder(
             tensor_device.toString(), std::move(shape)));
+    // Create sharded data placeholder, this will be used to
+    // hold the corresponding computation results.
+    if (tensor->sharding_spec()) {
+      auto sharding = tensor->sharding_spec();
+      if (!sharding->shape.has_value()) {
+        sharding->shape = tensor->shape();
+      }
+      handle = WrapXlaData(xla::ComputationClient::Get()->WrapDataShards(
+          {UnwrapXlaData(handle)}, GetVirtualDevice().toString(),
+          sharding->shape.value(), sharding->sharding));
+      tensor->data()->handle = handle;
```

Comment:
do we need to do it here? I thought we just need to push it to `tensor_data_vec` and it will get assigned later.

Reply:
I think ti is being down in 
https://github.com/pytorch/xla/blob/50141f2e19cedab7121d34f0921694402d686d9d/torch_xla/csrc/xla_graph_executor.cpp#L824-L827

so you can skip it here.

Reply:
Good catch, done.